articles/supervised-learning-algorithms/index.md

```diff
@@ -0,0 +1,158 @@
+
+    # Fit the classifier to the training data
+    knn.fit(X_train, y_train)
+
+    # Print the accuracy
+    print(knn.score(X_test,y_test))
+   ```
+
+   The output is shown below:
+   ```txt
+    0.9833333333333333
+   ```
+The accuracy of the classifier is 98.33%. This is tested on the test dataset. 98.33% is a good accuracy percentage, but the dataset is a simple one. 10 years ago, this number would have been a considered good one.
+
+2. Support Vector Machines(SVM): SVMs are maximum margin classifiers that are optimized to find an N-dimensional hyperplane in an N-dimensional space. The objective is to find the hyperplane that has the maximum margin from all the classes. Let's understand a few of the concepts and terminologies used in SVM.
+
+- Support Vector: Vectors that are closest to the hyperplane are called support vectors.
+- Margin: Margin is defined as the distance between data points and the hyperplane.
+- Hyperplane: The decision boundary which satisfies the maximum margin condition is called the hyperplane.
+
+The reasoning behind SVM is to find the hyperplane with the maximum distance from the support vectors. The hyperplane may be a linear decision boundary or a non-linear decision boundary. When dealing with non-linear planes, the dataset is projected into higher dimensions to create linear boundaries. For example, consider the following image.
+
+![nonlinearmapping](/engineering-education/supervised-learning-algorithms/svm.jpg)
+
+*[Image Source](https://www.researchgate.net/profile/PK_Varshney/publication/228712327/figure/fig3/AS:667775798353922@1536221565025/Mapping-nonlinear-data-to-a-higher-dimensional-feature-space-where-a-linear-separating.ppm)*
+
+The input space is transformed using kernels. SVM kernels are functions that take low-dimensional input space and transform them into higher dimensional space where the data is linearly separable. Some of the kernels commonly used are:
+
+- Linear Kernel
+- Polynomial Kernel
+- Radial Basis Function Kernel
+
+Let us look at implementing SVM using `sklearn`.
+
+   ```python
+    # Import necessary modules
+    from sklearn import svm
+    from sklearn.model_selection import train_test_split
+    from sklearn.datasets import load_digits
+    # Create feature and target arrays
+    digits = load_digits()
+    X = digits.data
+    y = digits.target
+
+    # Split into training and test set
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state=42, stratify=y)
+
+    # Create a SVM classifier
+    classifier = svm.SVC(kernel='poly')
+    classifier.fit(X_train, y_train)
+    print(classifier.score(X_test,y_test))
+   ```
+
+### Regression
+Regression algorithms are another subset of machine learning algorithms used to predict continuous numeric responses. As seen in an earlier example, predicting house rent given different factors is an example of regression. Let's look at the regression algorithm and use ridge regression as an example.
```

Comment:
? @lalith1403  - while reading this paragraph - I see that you mentioned ridge regression ? But then below you talk about linear regression? 
not sure if its the same thing - but that was a bit confusing

Reply:
Sorry about that... I meant linear regression.